File: 백준/Silver/1193. 분수찾기/분수찾기.py
import sys
input = sys.stdin.readline

X = int(input())

i = 0
total = 0

while total < X:
    i += 1
    total += i

start = (X - (total - i))

n = 1
m = i
if i % 2 == 0:
    for a in range(1, start):
        n += 1
        m -= 1
    print(f'{n}/{m}')
else:
    for a in range(i, start, -1):
        n += 1
        m -= 1
    print(f'{n}/{m}')

File: 백준/Silver/1193. 분수찾기/분수찾기.js
let fs = require('fs')
let input = fs.readFileSync("/dev/stdin").toString().trim()
// let input = fs.readFileSync("./EX.txt").toString().trim()

const X = parseInt(input)
let line = 0
let total = 0

while (total < X) {
    line += 1
    total += line
}

const start = (X - (total - line ))

n = 1
m = line

if (line % 2 == 0){
    for (let a = 1; a < start; a++) {
        n += 1
        m -= 1
    }
    console.log(`${n}/${m}`)
} else {
    for (let a = line; a > start; a--) {
        n += 1
        m -= 1
    }
    console.log(`${n}/${m}`)
}